django/urls/resolvers.py

```diff
@@ -657,6 +667,13 @@ def resolve(self, path):
                         sub_match_dict = {**kwargs, **self.default_kwargs}
                         # Update the sub_match_dict with the kwargs from the sub_match.
                         sub_match_dict.update(sub_match.kwargs)
+                        # assign the additional arguments in sub_match to
+                        # submatch_extra_dict and merge it with the extra_kwargs
+                        # from the sub_match
+                        sub_match_extra_dict = {
+                            **self.default_kwargs,
+                            **sub_match.extra_kwargs,
+                        }
```

Comment:
In my opinion, we do need access to extra_context args as ResolverMatch combined the kwargs captured from the URL-regex with the additional kwargs. And that's what was the problem statement referred to. Now when this combination occurs captured kwargs get overwritten with additional kwargs, if they both contain the same key.

So, because captured kwargs were getting overwritten, I provided additional access to them, and the because ticket statement was that to have additional access to extra context separately I provided access to additional kwargs as well.

Reply:
And, for `default_kwargs` I wasn't aware of the precedence of it that much, thanks for letting me know that. 👍

Reply:
for the part of the tests, I have added some changes and two tests in `tests/urlpatterns/tests.py` regarding `extra_kwargs`, though I can add one more test where captured kwargs are getting overwritten with additional kwargs.